sdk/tables/Azure.Data.Tables/samples/Sample6TransactionalBatch.md

```diff
@@ -0,0 +1,137 @@
+# Transactional Batches
+
+This sample demonstrates how to create and submit transactional batches for table entities.
+The Table service supports batch transactions on entities that are in the same table and belong to the same partition group.\
+Multiple Add, Update, Upsert, and Delete operations are supported within a single transaction.
+
+You will have needed to [create a table](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/samples/Sample1CreateDeleteTables.md) in the service in order to submit a transactional batch request.
+To get started, you'll need access to either a Storage or Cosmos DB account.
+
+## Create a `TableClient`
+
+A `TableClient` is needed to perform table-level operations like inserting and deleting entities within the table, so it is ideal for dealing with only a specific table. There are two ways to get a `TableClient`:
+- Call `GetTableClient` from the `TableServiceClient` with the table name.
+
+```C# Snippet:TablesSample1GetTableClient
+tableName = "OfficeSupplies1p2";
+var tableClient = serviceClient.GetTableClient(tableName);
+```
+
+- Create a `TableClient` with a SAS URI, an endpoint and `TableSharedKeyCredential`, or a connection string.
+
+```C# Snippet:TablesSample1CreateTableClient
+tableClient = new TableClient(
+    new Uri(storageUri),
+    tableName,
+    new TableSharedKeyCredential(accountName, storageAccountKey));
+```
+
+If you are not familiar with creating tables, refer to the sample on [creating and deleting tables](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/samples/Sample1CreateDeleteTables.md).
+
+## Adding entities with a transactional batch
+
+A common use case for batch operations is to add many entities to a table in bulk.
+
+```C# Snippet:BatchAdd
+// Create a list of 5 entities with the same partition key.
+string partitionKey = "BatchInsertSample";
+List<TableEntity> entityList = new List<TableEntity>{
+    new TableEntity(partitionKey, "01")
+    {
+        {"Product", "Marker" },
+        {"Price", 5.00 },
+        {"Brand", "Premium" }
+    },
+    new TableEntity(partitionKey, "02")
+    {
+        {"Product", "Pen" },
+        {"Price", 3.00 },
+        {"Brand", "Premium" }
+    },
+    new TableEntity(partitionKey, "03")
+    {
+        {"Product", "Paper" },
+        {"Price", 0.10 },
+        {"Brand", "Premium" }
+    },
+    new TableEntity(partitionKey, "04")
+    {
+        {"Product", "Glue" },
+        {"Price", 1.00 },
+        {"Brand", "Generic" }
+    },
+};
+
+// Create the batch.
+TableTransactionalBatch addBatch = client.CreateTransactionalBatch(partitionKey);
```

Comment:
nit:  `addBatch` confused me at first, since it sounds like a verb.  You may want to consider renaming to something that reads as more noun.

Reply:
What about the variable 'transaction' or 'addTransaction'?